Add params collections (#40978)
dotnet/docs: docs/csharp/methods.md

The following example defines a class named `Motorcycle` that contains five methods:

:::code language="csharp" source="snippets/methods/methods40.cs" id="snippet40":::

> The `Motorcycle` class includes an overloaded method, `Drive`. Two methods have the same name, but you must differentiate them by their parameter types.

Reviewer: ```suggestion
The `Motorcycle` class includes an overloaded method, `Drive`. Two methods have the same name, but are differentiated by their parameter types.
```

Reviewer: It's describing someone else's code, not code the reader would/could write.

:::code language="csharp" source="snippets/methods/byref106.cs" id="snippet106":::

A common pattern that uses by ref parameters involves swapping the values of variables. You pass two variables to a method by reference, and the method swaps their contents. The following example swaps integer values.

:::code language="csharp" source="snippets/methods/swap107.cs" id="snippet107":::

Passing a reference-type parameter allows you to change the value of the reference itself, rather than the value of its individual elements or fields.

### Parameter collections

Sometimes, the requirement that you specify the exact number of arguments to your method is restrictive. By using the `params` keyword to indicate that a parameter is a parameter collection, you allow your method to be called with a variable number of arguments. The parameter tagged with the `params` keyword must be a collection type, and it must be the last parameter in the method's parameter list.

> A caller can then invoke the method in either of four ways:

Reviewer: ```suggestion
A caller can then invoke the method in either of four ways for the `params` parameter:
```
